Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,265 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ARKit
+import SwiftUI
+import ArcGIS
+
+/// A scene view that provides an augmented reality table top experience.
+public struct TableTopSceneView: View {
+    /// The last portrait or landscape orientation value.
+    @State private var lastGoodDeviceOrientation = UIDeviceOrientation.portrait
+    @State private var arViewProxy = ARSwiftUIViewProxy()
+    @State private var sceneViewProxy: SceneViewProxy?
+    @State private var initialTransformation: TransformationMatrix? = nil
+    @State private var cameraController: TransformationMatrixCameraController
+    private let sceneViewBuilder: (SceneViewProxy) -> SceneView
+    private let configuration: ARWorldTrackingConfiguration
+    private var initialTransformationIsSet: Bool { initialTransformation != nil }
+    
+    /// Creates a table top scene view.
+    /// - Parameters:
+    ///   - anchorPoint: The location point of the ArcGIS Scene that is anchored on a physical surface.
+    ///   - translationFactor: The translation factor that defines how much the scene view translates
+    ///   as the device moves.
+    ///   - clippingDistance: Determines the clipping distance in meters around the camera. A value
+    ///   of `nil` means that no data will be clipped.
+    ///   - sceneView: A closure that builds the scene view to be overlayed on top of the
+    ///   augmented reality video feed.
+    /// - Remark: The provided scene view will have certain properties overridden in order to
+    /// be effectively viewed in augmented reality. Properties such as the camera controller,
+    /// and view drawing mode.
+    public init(
+        anchorPoint: Point,
+        translationFactor: Double,
+        clippingDistance: Double?,
+        @ViewBuilder sceneView: @escaping (SceneViewProxy) -> SceneView
+    ) {
+        self.sceneViewBuilder = sceneView
+        
+        let initialCamera = Camera(location: anchorPoint, heading: 0, pitch: 90, roll: 0)
+        let cameraController = TransformationMatrixCameraController(originCamera: initialCamera)
+        cameraController.translationFactor = translationFactor
+        cameraController.clippingDistance = clippingDistance
+        _cameraController = .init(initialValue: cameraController)
+        
+        configuration = ARWorldTrackingConfiguration()
+        configuration.worldAlignment = .gravityAndHeading
+        configuration.planeDetection = [.horizontal]
+    }
+    
+    public var body: some View {
+        ZStack {
+            ARSwiftUIView(proxy: arViewProxy)
+                .onDidUpdateFrame { _, frame in
+                    guard let sceneViewProxy else { return }
+                    updateLastGoodDeviceOrientation()
+                    sceneViewProxy.updateCamera(
```

Comment:
Since `sceneViewProxy.updateCamera` is now used outside of just the flyoverSceneView, we should move that extension into it's own file.

Reply:
Agreed, however theses errors occur when moving `updateCamera()` into a separate file:
<img width="875" alt="Screenshot 2023-09-26 at 11 51 59 AM" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/117859673/7be74f26-3842-4bc6-8114-6b0c7385616b">


<img width="873" alt="Screenshot 2023-09-26 at 11 53 22 AM" src="https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/117859673/3f8b3ccc-4355-459f-a9c6-4217a67feff3">

Reply:
I'm not seeing that. Can you try branch `ryan/tabletop2` please?

Reply:
Yes, is that branch published?

Reply:
The code compiles without errors on `ryan/tabletop2`. Thanks.

Reply:
I'm not sure what the difference was. Maybe an import statement?

Reply:
No it was because I placed the file under the components folder rather than the extensions folder.